clippy_lints/src/methods/extend_with_drain.rs

```diff
@@ -21,6 +21,8 @@ pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, recv: &Expr<'_>, arg:
         //check drain range
         if let src_ty_range = cx.typeck_results().expr_ty(drain_arg).peel_refs();
         if is_type_lang_item(cx, src_ty_range, LangItem::RangeFull);
+        //check if drain vec is a immutable type
+        let immutable= cx.typeck_results().expr_ty(drain_vec).is_mutable_ptr();
```

Comment:
```suggestion
        // check if drain vec is an immutable type
        let immutable = cx.typeck_results().expr_ty(drain_vec).is_mutable_ptr();
```

Reply:
I would also move this to line 19 where you already work with the type of `drain_vec`

```rs
let src_ty = cx.typeck_results().expr_ty(drain_vec);
let immutable = src_ty.is_mutable_ptr();
let src_ty = src_ty.peel_refs();
```

Reply:
Have you seen my comment above @Valentine-Mario?

Reply:
yes